contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -75,19 +88,39 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
     winner_token_id: Uint256, loser_token_id: Uint256
 ) {
     alloc_locals;
-    // Only Combat
+    // Only combat
+    Module.only_approved();
 
-    // TODO: Fix auth issue
-    // Module.only_approved()
+    let (controller) = Module.controller_address();
 
     let (current_relic_owner) = get_current_relic_holder(loser_token_id);
 
     let (is_equal) = uint256_eq(current_relic_owner, loser_token_id);
 
+    let (old_owner_token_id) = storage_relic_holder.read(current_relic_owner);
+
+    // If old owner exists, update array
+    let (check_exists) = uint256_lt(Uint256(0, 0), old_owner_token_id);
+
+    let (remove_victory_relics: Uint256*) = alloc();
+    let (remove_order_relics: Uint256*) = alloc();
+
     // Capture Relic if owned by loser
     // If Relic is owned by loser, send to victor
     if (is_equal == TRUE) {
-        _set_relic_holder(loser_token_id, winner_token_id);
+        if (check_exists == TRUE) {
+            let (loser_relics_len) = owned_relics_len.read(loser_token_id);
+            assert remove_victory_relics[0] = loser_token_id;
+            loop_remove_relic(0, 0, loser_relics_len, loser_token_id, 1, remove_victory_relics);
```

Comment:
I'm a bit uneasy about this part. Imagine this function getting called with `loser_relics_len` equal to 0. Then the `check` inside it will be `TRUE` and `0 - 1` (a "very big number" in Cairo) will get written in the storage var.

I think the scenario can happen. Say we have realms A, B & C, A & C in the same order and the game is in initial state (storage vars are zeroed). A fights B, wins. B fights A, wins. C fights B, wins. C unstakes (so relics are returned). A fights B, wins. I think in this scenario both checks will be TRUE (so we get to this part), but also `loser_relics_len` will be 0 because of the C vs. B fight (where B lost). Or do I have it wrong?

Reply:
`loser_relics_len` will not be 0 as `loop_return_relics` calls `_set_relic_holder` which increments B's count on the return. i.e we are taking into account that B has gained a relic when C unstaked (it was returned)

I've added a stress test for this, and read the relics length of 'Realm B' after the scenario asserting it is 0.